Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/OnCall.hs

```diff
@@ -0,0 +1,55 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+
+module Domain.Action.UI.OnCall
+  ( OnCallReq (..),
+    logOnCall,
+    sendCallDataToKafka,
+  )
+where
+
+import Data.Aeson
+import qualified Domain.Types.Merchant as Merchant
+import qualified Domain.Types.Person as Person
+import qualified Domain.Types.Ride as Ride
+import Kernel.Beam.Functions (runInReplica)
+import Kernel.External.Encryption
+import Kernel.Prelude
+import Kernel.Storage.Esqueleto.Config (EsqDBReplicaFlow)
+import qualified Kernel.Types.APISuccess as APISuccess
+import Kernel.Types.Id
+import Kernel.Utils.Common
+import Lib.SessionizerMetrics.Types.Event
+import qualified Storage.Queries.Ride as QRide
+import Tools.Error
+import Tools.Event
+
+data OnCallReq = OnCallReq
+  { rideId :: Id Ride.Ride,
+    callType :: Text
+  }
+  deriving (Generic, ToJSON, FromJSON, ToSchema)
+
+logOnCall :: (EsqDBFlow m r, EncFlow m r, EsqDBReplicaFlow m r, EventStreamFlow m r) => (Id Person.Person, Id Merchant.Merchant) -> OnCallReq -> m APISuccess.APISuccess
+logOnCall (personId, _) OnCallReq {..} = do
+  sendCallDataToKafka Nothing rideId callType Nothing Nothing User (Just personId)
+  pure APISuccess.Success
+
+sendCallDataToKafka :: (EsqDBFlow m r, EncFlow m r, EsqDBReplicaFlow m r, EventStreamFlow m r) => Maybe Text -> Id Ride.Ride -> Text -> Maybe Text -> Maybe Text -> EventTriggeredBy -> Maybe (Id Person.Person) -> m ()
+sendCallDataToKafka vendor rideId callType callSid callStatus triggeredBy personId = do
+  ride <- runInReplica $ QRide.findById rideId >>= fromMaybeM (RideDoesNotExist rideId.getId)
+  unless (isValidRideStatus (ride.status)) $ throwError $ RideInvalidStatus "The ride has already started."
+  triggerExophoneEvent $ ExophoneEventData vendor callType rideId callSid callStatus ride.merchantId triggeredBy personId
```

Comment:
what is personId in this ?

Reply:
The Id of the person who sent the request